File: solutions/cpp/leap/1/leap.cpp
#include "leap.h"

namespace leap
{
	bool is_leap_year(int year)
	{
		bool res;

		res = false;
		if ((year % 4) == 0)
		{
			res = true;
			if ((year % 100) == 0)
				res = false;
			if ((year % 400) == 0)
				res = false;
		}
		return res;
	}

} // namespace leap

File: solutions/free-pascal/leap/1/Leap.pas
unit Leap;

{$mode ObjFPC}{$H+}

interface

function LeapYear(const year : integer) : boolean;

implementation

uses SysUtils;

function LeapYear(const year : integer) : boolean;
begin
  if (year mod 4 = 0) then
    begin
      if ((year mod 100)=0) then
        if ((year mod 400)<>0) then
          exit(false);
      exit(true);
    end;
  LeapYear:= false;
end;
end.
